torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -1093,7 +1121,9 @@ bool FunctionSignature::parse(PyObject* self, PyObject* args, PyObject* kwargs,
     // tracer is enabled. This behavior easily leads to ambiguities, and we
     // should avoid having complex signatures that make use of it...
     } else if (allow_varargs_intlist && arg_pos == 0 && !is_kwd &&
-               THPUtils_checkIndex(obj)) {
+                ((param.type_ == 
+                ParameterType::SYM_INT_LIST && is_int_or_symint(obj)
+               ) || all_ints_in_tuple(args))) {
```

Comment:
I'm willing to live with this now, but we need a lot more commentary about what is going on here. My ideal solution is I would like to avoid having to traverse the entire python tuple when we are doing overload resolution ( the `all_ints_in_tuple` call here). I think the easiest way to achieve this, which I alluded to in chat, is to always prefer the symbolic integer overload and detect whether or not it's all integers when you're actually packing the integer/symbolic integer vector. This makes the downstream code a little more complicated because the integer list overload is actually dead, but I suppose we could do a refactor where we just drop the integer list overload, and have `symintlist` return either an int vector or a symint vector, and then call the appropriate C++ logic from there.

Reply:
I'd also prefer to eventually drop the `int` overload. CUDA/CPU the symint overloads already treat symints as concrete ints, and LTC needs to handle both.
Unfortunately, I think this PR already has a pretty big blast radius, so hopefully we can follow up in the next one?

Reply:
yes